PUMI/engine.py

```diff
@@ -464,3 +466,76 @@ def wrapper(name, bids_dir, subjects=None, base_dir='.', sink_dir=None, qc_dir=N
             return wf
 
         return wrapper
+
+class BidsApp:
+
+    def __init__(self, pipeline, name, bids_dir=None, output_dir=None, analysis_level=None, participant_label=None,
+                 working_dir=None, run_args=None, description=None, **kwargs):
+
+        if description is None:
+            self.parser = argparse.ArgumentParser()
+        else:
+            self.parser = argparse.ArgumentParser(description=description)
+
+        self.parser.add_argument('--bids_dir', required=False, help='Root directory of the BIDS-compliant input dataset.')
```

Comment:
Can't we just set required to True by default?

Reply:
That would be possible, but then the parameter really has to be passed via the cli. It's not possible that we set this parameter to True and then the developer sets the bids_dir in the BidsApp constructor for the end user. But it's possible that both specify a bids_dir and we select one of the inputs. If this is not a problem, we can change this.

Why this is not possible without problems: The ArgumentParser gets the inputs via sys.argv. We would have to manipulate the environment variable sys.argv at runtime for ArgumentParser not to throw an error if the user does not specify the bids_dir, which is not recommended.